services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -93,11 +93,17 @@ public RealmModel getRealmByName(String name) {
     }
 
     public RealmModel createRealm(String name) {
-        return createRealm(name, name);
+        return createRealm(null, name);
     }
 
     public RealmModel createRealm(String id, String name) {
-        if (id == null) {
+        // map storage requires UUIDs
+        if (Profile.isFeatureEnabled(Profile.Feature.MAP_STORAGE)) {
+            if (!KeycloakModelUtils.isValidUUID(id)) {
+                id = KeycloakModelUtils.generateId();
+            }
+        }
```

Comment:
I think the configuration option would be nice. I agree with the proposed solution to have it by default set to false and throw an exception when set to true. On top of that I think we should add a warning which would alert a user that the id of realm is going to be changed.

Reply:
Incorporated all these suggestions. For now I'm using a config property from the map storage, but it can be switched into something else, like a system property.

Reply:
Sorry that I'm late for this discussion: For the LDAP store, all IDs are determined by the LDAP server as GUIDs, so all ids passed in by Keycloak will be replaced in the returned entity with a GUID. 

The automatic conversion is therefore correct IMHO. Thinking about configuration, there having an automatic conversion is what is expected from a map store as it is allowed to treat any passed in ID as a mere suggestion. 

I would even go so far to make it non-configurable and also to not log a warning unless you have a scenario in mind where the conversion could break an existing behavior.